app/src/main/java/app/revanced/integrations/youtube/patches/components/KeywordContentFilter.java

```diff
@@ -0,0 +1,156 @@
+package app.revanced.integrations.youtube.patches.components;
+
+import static app.revanced.integrations.shared.StringRef.str;
+import static app.revanced.integrations.youtube.ByteTrieSearch.convertStringsToBytes;
+
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.ByteTrieSearch;
+import app.revanced.integrations.youtube.settings.Settings;
+
+/**
+ * <pre>
+ * Allows hiding home feed and search results based on keywords and/or channel names.
+ *
+ * Limitations:
+ * - Searching for a keyword phrase will give no search results
+ * - Filtering a channel name can still show Shorts from that channel in the search results
+ * - Some layout component residue will remain, such as the video chapter previews for some search results.
+ * - Keywords are case sensitive, but some casing variation is manually added.
+ *   (ie: "mr beast" automatically filters "Mr Beast" and "MR BEAST").
+ */
+@SuppressWarnings("unused")
+@RequiresApi(api = Build.VERSION_CODES.N)
+final class KeywordContentFilter extends Filter {
+
+    /**
+     * Minimum keyword/phrase length to prevent excessively broad content filtering.
+     */
+    private static final int MINIMUM_KEYWORD_LENGTH = 3;
+
+    /**
+     * Substrings that are always first in the path.
+     */
+    private final StringFilterGroup startsWithFilter = new StringFilterGroup(
+            Settings.HIDE_KEYWORD_CONTENT,
+            "home_video_with_content.eml",
+            "search_video_with_context.eml",
+            "related_video_with_content",
+            "compact_video.eml",
```

Comment:
grid_video_wrapper.em appears in channels which features a list of videos. The filter can be used to block keywords in the title.

Reply:
> Do users want to hide videos for a channel they intentionally navigated to?

In some scenarios, this can be useful. For example, they talk about many subjects if you want to filter for profanity or a specific topic, such as in the LinusTechTips channel.

In other scenarios, that may not be wanted, such as when you want to filter a specific channel entirely, such as Mr Beast, but occasionally want to manually check their channel.

That could be solved by allowing the user to choose the filters via an additional input preference. Alternatively, a couple of switches, such as a "Filter channel videos" switch, could be added for such cases.